### PR TITLE
Json service

### PR DESCRIPTION
**Pull request by mortenn**
Work in progress

### CODE REVIEW

File: src/API/Request.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+	 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+	 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+	 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+	 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+	 * SOFTWARE.
+	 */
+
+	namespace KrameWork\API;
+
```
Comment (Kruithne) [inline src/API/Request.php:26]: Missing class header (package, author, etc).

File: src/API/CacheAwareService.php [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	class ServiceResponse implements \JsonSerializable
+	{
+		/**
+		 * ServiceResponse constructor.
+		 *
+		 * @api __construct
+		 * @param bool $success
+		 * @param mixed $data
+		 */
+		public function __construct(bool $success = true, $data = []) {
```
Comment (KalleZ) [inline src/API/CacheAwareService.php:43]: $data should be type-hinted to array as its used as an array later on, unless its an object that implements \ArrayAccess

**Comment by Kruithne** [comment]
What's the verdict on this addition; do you still plan to develop it or are we scrapping it?

**Comment by KalleZ** [comment]
If developed further, remember to prefix global function calls with \, I spot a fair few things there

**Comment by mortenn** [comment]
I am fairly certain I will be scrapping this branch entirely and do something new to cover the same topic

**Pull request closed without merging**